skipper.go

```diff
@@ -985,7 +990,16 @@ func listen(o *Options, mtr metrics.Metrics) (net.Listener, error) {
 	}
 
 	if !o.EnableTCPQueue {
-		return net.Listen("tcp", o.Address)
+		ln, err := net.Listen("tcp", o.Address)
+		if o.ProxyProtocolEnable {
```

Comment:
I think you need to add test cases in the `skipper_test.go` for your proxy protocol support case. Check if `skipper.go` does include any tests of this sort for `EnableTCPQueue` option, if not then I guess you can leave as it is, or try to test it properly :).